macos/FluentUI/Button/Button.swift

```diff
@@ -64,7 +64,7 @@ open class Button: NSButton {
 
 	/// Initializes a Fluent UI Button with no title or image, and with default formatting
 	@objc public convenience init() {
-		self.init()
+		self.init(title: nil, image: nil, imagePosition: .imageLeading, format: ButtonFormat())
```

Comment:
I just wanna say lol

Reply:
I learned about this craziness today and had to work around it for Link.  A convenience initializer must call a designated initializer, right?  Why would it default to calling itself?

Reply:
Oh I thought this was originally from me, my bad.  I'm not sure what the "convenience" keyword does, I jsut saw it on other code

Reply:
[Designated Initializers and Convenience Initializers - The Swift Programming Language](https://docs.swift.org/swift-book/LanguageGuide/Initialization.html#ID216)

Reply:
Quoting from the swift doc:
Rule 1
A designated initializer must call a designated initializer from its immediate superclass.
Rule 2
A convenience initializer must call another initializer from the same class.
Rule 3
A convenience initializer must ultimately call a designated initializer.

I think the key part is Rule 2, where self.init() was allowed to call another convenience initializer (itself), but it never reached to the designated initializer.